src/lucid/lucid.ts

```diff
@@ -256,11 +257,24 @@ export class Lucid {
   }
 
   selectWallet(api: WalletApi): Lucid {
+    const getAddressHex = async () => {
+      const [addressHex] = await api.getUsedAddresses();
+
+      if (addressHex) return addressHex;
+
+      const [unusedAddressHex] = await api.getUnusedAddresses();
+
+      return unusedAddressHex;
+    }
+
     this.wallet = {
-      address: async (): Promise<Address> =>
-        C.Address.from_bytes(
-          fromHex((await api.getUsedAddresses())[0]),
-        ).to_bech32(undefined),
+      address: async (): Promise<Address | null> => {
```

Comment:
We can safely assume now that we get an address back. If no address is returned then the wallet you are connected to is broken. So it should stay `Promise<Address>` and you don't need to make all these other changes.

Reply:
Right! I will fix it.